lte/gateway/python/magma/pipelined/app/conntrack.py

```diff
@@ -0,0 +1,231 @@
+
+        # Match all new connections
+        match = MagmaMatch(eth_type=ether_types.ETH_TYPE_IP,
+                           ct_state=(self.CT_NEW | self.CT_TRK,
+                                     self.CT_NEW | self.CT_TRK))
+        actions = [parser.NXActionCT(
+            flags=0x1,
+            zone_src=None,
+            zone_ofs_nbits=0,
+            recirc_table=self.connection_event_table,
+            alg=0,
+            actions=[]
+        )]
+        flows.add_drop_flow(datapath, self.conntrack_scratch,
+                            match, actions,
+                            priority=flows.DEFAULT_PRIORITY)
+
+        # Match tcp terminations (fin)
+        match = MagmaMatch(eth_type=ether_types.ETH_TYPE_IP,
+                           ip_proto=IPPROTO_TCP,
+                           tcp_flags=(0x1,0x1),
+                           ct_state=(self.CT_EST | self.CT_TRK,
+                                     self.CT_EST | self.CT_TRK))
+        actions = [parser.NXActionCT(
+            flags=0x0,
+            zone_src=None,
+            zone_ofs_nbits=0,
+            recirc_table=self.connection_event_table,
+            alg=0,
+            actions=[]
+        )]
+        flows.add_drop_flow(datapath, self.conntrack_scratch,
+                            match, actions,
+                            priority=flows.DEFAULT_PRIORITY)
+        # match tcp fin
+        match = MagmaMatch(eth_type=ether_types.ETH_TYPE_IP,
+                           ip_proto=IPPROTO_TCP,
+                           tcp_flags=(0x1, 0x1),
+                           ct_state=(self.CT_TRK | self.CT_INV, self.CT_TRK | self.CT_INV))
+        # flags 0 is nothing, 1 is commit
+        actions = [parser.NXActionCT(
+            flags=0x0,
+            zone_src=None,
+            zone_ofs_nbits=0,
+            recirc_table=self.connection_event_table,
+            alg=0,
+            actions=[]
+        )]
+        flows.add_drop_flow(datapath, self.conntrack_scratch,
+                            match, actions,
+                            priority=flows.DEFAULT_PRIORITY)
+
+        # match tcp rst
+        match = MagmaMatch(eth_type=ether_types.ETH_TYPE_IP,
+                           ip_proto=IPPROTO_TCP,
+                           tcp_flags=(0x4, 0x4),
+                           ct_state=(self.CT_EST | self.CT_TRK, self.CT_EST | self.CT_TRK))
+        # flags 0 is nothing, 1 is commit
+        actions = [parser.NXActionCT(
+            flags=0x0,
+            zone_src=None,
+            zone_ofs_nbits=0,
+            recirc_table=self.connection_event_table,
+            alg=0,
+            actions=[]
+        )]
+        flows.add_drop_flow(datapath, self.conntrack_scratch,
+                            match, actions,
+                            priority=flows.DEFAULT_PRIORITY)
+
+        inbound_match = MagmaMatch(eth_type=ether_types.ETH_TYPE_IP,
+                                   direction=Direction.IN)
```

Comment:
can you squash multiple patches in single commit, its tricky to review patch when first patch adds code and second removes it, let me know if there is way to look at final code on github.

Reply:
Not sure If i follow, when using github UI you can see the final code. https://github.com/magma/magma/pull/2191/files

Reply:
ok, I was looking at commits. I generally pull files on my laptop.